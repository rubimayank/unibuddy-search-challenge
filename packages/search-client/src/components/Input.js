import React from 'react';
import styled from 'styled-components';
import {
  FiX,
} from 'react-icons/fi';

const Wrapper = styled.div`
  width: 100%;
  height: 40px;
  position: relative;
`;

const IconWrapper = styled.div`
  position: absolute;
  z-index: 1;
  height: 100%;
  padding: 8px;
  display: flex;
  align-items: center;
`;

const ClearButton = styled.button`
  position: absolute;
  z-index: 1;
  height: 100%;
  margin: 0;
  padding: 8px;
  border: 0;
  background: none;
  display: flex;
  align-items: center;
  right: 0;
  top: 0;
`;

const StyledInput = styled.input`
  width: 100%;
  height: 100%;
  flex: 1;
  padding: 8px 24px;;
  z-index: 0;
  border: 1px solid gray;
  border-radius: 4px;
`;

function Input({icon, onClear, ...inputProps}) {
  return (
    <Wrapper>
      <IconWrapper>{icon}</IconWrapper>
      <StyledInput {...inputProps} />
      {onClear && (
        <ClearButton onClick={onClear} title='Clear'>
          <FiX />
        </ClearButton>
      )}
    </Wrapper>
  );
}

export default Input;
