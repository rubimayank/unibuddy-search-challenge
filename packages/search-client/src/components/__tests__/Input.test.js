import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Input from '../Input';

describe('input', () => {
  it('it should render', () => {
    const snap = renderer.create(<Input />).toJSON();
    expect(snap).toMatchSnapshot();
  });

  it('should render with icon', () => {
    const snap = renderer.create(<Input icon={<i>i</i>}/>).toJSON();
    expect(snap).toMatchSnapshot();
  });

  it('should render with clear button', () => {
    const snap = renderer.create(<Input onClear={()=>{}}/>).toJSON();
    expect(snap).toMatchSnapshot();
  });

  it('should call onClear', () => {
    let foo = 'bar';
    const wrapper = renderer.create(<Input onClear={()=>{ foo=''; }} />).root;
    const clearBtn = wrapper.findByType('button');
    clearBtn.props.onClick();
    expect(foo).toEqual('');
  });
});
