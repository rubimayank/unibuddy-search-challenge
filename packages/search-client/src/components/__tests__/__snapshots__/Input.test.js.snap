// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input it should render 1`] = `
.c0 {
  width: 100%;
  height: 40px;
  position: relative;
}

.c1 {
  position: absolute;
  z-index: 1;
  height: 100%;
  padding: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 100%;
  height: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 8px 24px;
  z-index: 0;
  border: 1px solid gray;
  border-radius: 4px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <input
    className="c2"
  />
</div>
`;

exports[`input should render with clear button 1`] = `
.c0 {
  width: 100%;
  height: 40px;
  position: relative;
}

.c1 {
  position: absolute;
  z-index: 1;
  height: 100%;
  padding: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  position: absolute;
  z-index: 1;
  height: 100%;
  margin: 0;
  padding: 8px;
  border: 0;
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  right: 0;
  top: 0;
}

.c2 {
  width: 100%;
  height: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 8px 24px;
  z-index: 0;
  border: 1px solid gray;
  border-radius: 4px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <input
    className="c2"
  />
  <button
    className="c3"
    onClick={[Function]}
    title="Clear"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </button>
</div>
`;

exports[`input should render with icon 1`] = `
.c0 {
  width: 100%;
  height: 40px;
  position: relative;
}

.c1 {
  position: absolute;
  z-index: 1;
  height: 100%;
  padding: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 100%;
  height: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 8px 24px;
  z-index: 0;
  border: 1px solid gray;
  border-radius: 4px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <i>
      i
    </i>
  </div>
  <input
    className="c2"
  />
</div>
`;
